packages/discovery-provider/src/queries/get_collection_library.py

```diff
@@ -215,6 +215,49 @@ def _get_collection_library(args: GetCollectionLibraryArgs, session):
         user_id,
     )
 
+    # exclude hidden playlists and hidden albums which were not previously purchase by current user
+    playlists = list(
+        filter(
+            lambda playlist: not playlist["is_private"]
+            or (
+                collection_type == CollectionType.album and playlist["access"]["stream"]
```

Comment:
Does this mean "previously purchased"? I don't think it does? Wouldn't this be true for free private albums?

Reply:
fixed this